### DISCUSSION TITLE
Housing for PCB V2B (with 2021 St. Moritz spa)

### DISCUSSION THREAD

**Discussion by MacGyver1979** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#538] [Show and tell]
Hi everyone,

first of all a big thank you to @visualapproach for the great work!! Hope you enjoy the coffee :-) I´d like to give feedback to my application here.
Even though I have very few experience with these kind of projects I managed to get it working for my spa. Ordered the PCB at OSHWlab + ESP, LLC and JST connectors from Amazon ... and that's it.

Here's my configuration:
- PCB V2B
- Spa model "St. Moritz" from 2021
- Square pump model
- 6 pin connector
- MQTT connection to iobroker on PI4

I didn't need any further adaptions. Works like from the scratch! 
Maybe this data could be added in the documentation...?

Additionally, I am working on a 3D printed housing for PCB V2B. Not perfectly finished yet (needs some rework), but still it fits! 
Here's my first try:

![grafik](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/114698940/7ad3b460-18bd-4266-b6cc-9c3d9cfa8e09)
![grafik](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/114698940/4f7cee74-0169-4f89-bbcb-4dd3e6431efc)

I will post any news here as soon as the 3D is ready to use.

**Comment by steffzbl** [comment]
Hi, is the case still work in progress? i'd like to have a copy of that, it looks nice!

**Reply by MacGyver1979** [reply]
It's still not really finished, but I used the first version it after some rework. I had to cut the PCB fixation ribs a little bit.
Also the wall thickness is VERY strong (4mm). 3mm should be enough. 
Later on I want to add a fixation in the pump. Don't know how yet...

Unfortunately I don't have a 3D printer myself, so I always have to ask a friend to print it. That's why it takes me a long time ;-)
Nevertheless you might try the 3mm version, if you have a printer yourself:

[Housing_V1.1.zip](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/files/12455710/Housing_V1.1.zip)

Please let me know if it fits for you!

**Comment by BirbByte** [comment]
First off I'll say this design is really nice and would provide some protection to the main PCB & components, and it looks good too!

I've tried printing this case in PETG and using with the V2B PCB, though my controls began acting strange. I found that I couldn't toggle any settings (e.g. pump, air bubbles, heating mode) and even the touch control panel for the spa had the same problems, not passing any inputs through.. though I could still access the web interface. Shortly after removing the printed housing, it began working again. I do have my JST cables soldered directly to the far right edge (H1B & H2B).

I suspect that because the print is touching the soldered connections that perhaps somehow it's shorting out on the input (DSP) side, or putting force on the solder joint and making it weak.

I've made an edit of the design above (V1.1), I've changed just the bottom piece to include standoffs that go into the holes in the corners of the PCB. The attached file includes the original top piece, and the remixed bottom piece.

[Housing_V1.1_Remix.zip](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/files/13986901/Housing_V1.1_Remix.zip)

This new design is a tight fit, the spacing is probably out by 1mm or so. If you push the PCB slightly firmly (think inserting RAM into motherboard kind of force), it should press into place and will hold the board quite snug.


After all that I've found my board is more reliable, however at the end of the day I think my issue is possibly a bad solder join on one of the JST connectors. This design in my opinion is a bit better since it isn't placing any pressure on the solder joints now

**Reply by dannywerd** [reply]
Thanks for your contribution sharing the housing print. I will print it and give it a try.
